onnxruntime/core/providers/cuda/tensor/split_impl.cu

```diff
@@ -157,5 +157,122 @@ Status SplitImpl(cudaStream_t stream, const size_t element_size, const int block
   return Status::OK();
 }
 
+template <typename T>
+__device__ __forceinline__ void _vec_copy_data(
+    void* output_ptr,
+    const void* input_ptr,
+    const int64_t size_in_byte) {
+  auto output_ptr_vec = reinterpret_cast<T*>(output_ptr);
+  auto input_ptr_vec = reinterpret_cast<const T*>(input_ptr);
+  auto new_size = size_in_byte / sizeof(T);
+  for (int i = threadIdx.x; i < new_size; i += blockDim.x) {
+    output_ptr_vec[i] = input_ptr_vec[i];
+  }
+  auto remain = size_in_byte % sizeof(T);
```

Comment:
If we select proper T that `size_in_bytes % sizeof(T) == 0`, we do not need the remaining code here. See another comment.